astropy/__init__.py

```diff
@@ -231,10 +231,8 @@ def _rollback_import(message):
         # exception logger before we raise the import error:
         _teardown_log()
 
-        # Roll back any astropy sub-modules that have been imported thus
-        # far
-
-        for key in list(sys.modules):
+        # Roll back any astropy sub-modules that have been imported thus far
+        for key in sys.modules.keys():
```

Comment:
The reason the list was here is that we're deleting modules while iterating over them; hence, we need to be sure that `.keys()` doesn't change during the loop.  See https://stackoverflow.com/questions/5384914/how-to-delete-items-from-a-dictionary-while-iterating-over-it

Reply:
I think I missed looking in that angle. Anyway, thanks for the review @mhvk , appreciate it 👍.